### PR TITLE
Tested all functions in calf.py 

### PR DESCRIPTION
**Pull request by 0luchi**
- Tested all functions in calf.py to improve coverage to 100%

### CODE REVIEW

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (JoeWaddell) [inline tests/animal_module_tests/test_animal_life_cycle.py:1525]: Missing docstring here (and in others), the format should mimic the others in this file, so it tells the reader which method/function it's testing, and which file it came from. Both help with duplicate/similar method names, and also CTRL+SHIFT+F searching for method names.

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (JoeWaddell) [inline tests/animal_module_tests/test_animal_life_cycle.py:1469]: We'll need to add a test here for the `__init__` call as well.

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (JoeWaddell) [inline tests/animal_module_tests/test_animal_life_cycle.py:1560]: Looks like line 75 in calf.py isn't covered here: I think it'll require another line in the paramaterized args at the top, but it might be tricky.

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (JoeWaddell) [inline tests/animal_module_tests/test_animal_life_cycle.py:1584]: Here you could hardcode args values, and then check that the calf correctly had those applied: e.g.
`test_args = {'birth_weight': 100, 'body_weight': 150, 'wean_weight': 101, 'mature_body_weight': 300}`
then after the act: `assert calf.birth_weight == 100`...etc

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (JoeWaddell) [inline tests/animal_module_tests/test_animal_life_cycle.py:1646]: By the way, nice job making these test names unique!

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (JoeWaddell) [inline tests/animal_module_tests/test_animal_life_cycle.py:1688]: One more set of checks here to assert that the wean_weight and daily_growth are set properly would round this out nicely!

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (JoeWaddell) [inline tests/animal_module_tests/test_animal_life_cycle.py]: The act here isn't calling calc_manure_excretion, but the calc_base_manure method in animal_base.

**Review by JoeWaddell — CHANGES_REQUESTED** [review]
Looking good, it needs a few more assertions and docstrings, but otherwise it's great to see the coverage jump up for calf.py
Nice work!

**Review by JoeWaddell — APPROVED** [review]
Nice job! LGTM: good expansion of test coverage for calf.py 

In future updates we'll likely move these methods to their own test file, but given that there are a few unit test PRs open, best to wait until those are merged.

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (Isgotkowitz) [inline tests/animal_module_tests/test_animal_life_cycle.py]:
```suggestion
```

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (Isgotkowitz) [inline tests/animal_module_tests/test_animal_life_cycle.py]:
```suggestion
```
Birth weight is checked before getting into the if/else block.

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (Isgotkowitz) [inline tests/animal_module_tests/test_animal_life_cycle.py:1482]: This method achieves good coverage, however it is doing more than just testing `__init__`. It is also indirectly testing the methods `assign_calf_values()` and `init_values()`. Ideally a unit test would not be doing this, though an integration test would. Unfortunately mocking methods that are called in `__init__` is (in my experience) a pain and this test works well, so it is not worth rewriting it in this case.

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (Isgotkowitz) [inline tests/animal_module_tests/test_animal_life_cycle.py]:
```suggestion
```

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (Isgotkowitz) [inline tests/animal_module_tests/test_animal_life_cycle.py]: Incrementing `calf.days_born` and asserting `calf.days_born == wean_day` is duplicating/undoing a lot of logic that is being tested. A better approach would be defining the expected return value in the test parameters, then checking that what you have defined as your expectation matches what you are actually seeing in the test behavior.

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (Isgotkowitz) [inline tests/animal_module_tests/test_animal_life_cycle.py]: Daily growth is set whether or not the calf is weaned, but in this test it is only checked if the calf is not weaned. Also, this is asserting that when a calf is weaned it doesn't grow at all, which is not what the code does.

File: tests/animal_module_tests/test_animal_life_cycle.py
Comment (Isgotkowitz) [inline tests/animal_module_tests/test_animal_life_cycle.py]:
```suggestion
```

**Review by Isgotkowitz — APPROVED** [review]
Nice tests Oluchi. I noticed `test_update_calf()` in the testing logic doesn't seem to line up with the method logic. There are also some spots in `test_init_calf()` that should be cleaned up that I left comments on. Other than those places though, I think things are solid.

**Pull request merged**
